src/vcpkg/commands.depend-info.cpp

```diff
@@ -452,25 +457,34 @@ namespace vcpkg
 
         if (strategy.format == DependInfoFormat::Tree)
         {
-            Util::sort(depend_info, reverse);
-            auto first = depend_info.begin();
-            std::string features = Strings::join(", ", first->features);
-
-            if (strategy.show_depth)
+            std::set<std::string> printed;
+            for (auto&& info : depend_info)
             {
-                msg::write_unlocalized_text(Color::error, fmt::format("({})", first->depth));
-            }
+                if (info.depth != 0) continue;
 
-            msg::write_unlocalized_text(Color::success, first->package);
-            if (!features.empty())
-            {
-                msg::write_unlocalized_text(Color::warning, "[" + features + "]");
+                std::string features = Strings::join(", ", info.features);
+
+                if (strategy.show_depth)
+                {
+                    msg::write_unlocalized_text(Color::error, "(0)"); // legacy
+                }
```

Comment:
Entirely pointless, but kept for compatibility: output depth at top level tree (actually digraph now) nodes.
(Suggesting removal.)